### PR TITLE
fix(quantic):  Tooltip missing on long recent queries/suggestions in quantic searchbox

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5602](https://coveord.atlassian.net/browse/SFINT-5602)

## ISSUE:
- When we have long recent queries or long suggestions, there was currently no tooltip displayed on hover. This is both bad for a user who wants to see the whole suggestion and for accessibility reasons.
image

## FIX:

- Added a `title` tag to the element containing the text of the recent query / query suggestion. This displays a tooltip on hover over the text value.

## DEMO:
image

[SFINT-5602]: https://coveord.atlassian.net/browse/SFINT-5602?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.html [hunk 1]
```diff
@@ -25,8 +25,8 @@
                                     alternative-text={option.iconTitle}></lightning-icon>
                             </span>
                             <span class="slds-media__body slds-grid">
-                                <lightning-formatted-rich-text class="slds-truncate"
-                                    value={option.value}></lightning-formatted-rich-text>
+                                <lightning-formatted-rich-text class="slds-truncate suggestion-option__text"
+                                    value={option.value} title={option.value}></lightning-formatted-rich-text>
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.html:29]:
```suggestion
                                <lightning-formatted-rich-text class="slds-truncate" data-cy="suggestion-option__text"
                                    value={option.value} title={option.value}></lightning-formatted-rich-text>
```

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ const selectors = {
   searchBoxComboBox: '.slds-combobox_container .slds-combobox',
   searchBoxSearchIcon: '.searchbox__search-icon',
   suggestionOption: '[data-cy="suggestions-option"]',
+  suggestionOptionText: '.suggestion-option__text',
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:39]:
```suggestion
  suggestionOptionText: '[data-cy="suggestion-option__text"]',
```

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js [hunk 3]
```diff
@@ -219,6 +220,18 @@ describe('c-quantic-search-box-input', () => {
...
+              suggestionsList.shadowRoot.querySelectorAll(
+                selectors.suggestionOptionText
+              );
+            const suggestionsLength = mockSuggestions.length;
+
+            expect(suggestionOptionLabels).not.toBeNull();
+            expect(suggestionOptionLabels.length).toEqual(suggestionsLength);
+            expect(suggestionOptionLabels[suggestionsLength - 1].title).toEqual(
+              mockSuggestions[suggestionsLength - 1].value
+            );
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:234]: You can't just check the length or the title of the last one.


```suggestion
mockSuggestions.forEach((suggestion, idx) => {
  expect(suggestionOptionsLabels[idx].title).toEqual(suggestion.value);
});
```

**Review by erocheleau — CHANGES_REQUESTED** [review]
Needs to change to cypress selector,

also better expectation.

**Comment by mmitiche** [comment]
> _Note: The `<b></b>` tag in the tooltip is caused bny the issue raised in SFINT- 5601 and will be fixed there._


Actually no, this issue is not caused because of that, you need to pass `option.rawValue` as the title instead of `option.value` cause value is the formatted `value` and `rawValue` is the raw value that we need in this case.

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js [hunk 4]
```diff
@@ -259,6 +277,24 @@ describe('c-quantic-search-box-input', () => {
...
+            const suggestionsAndRecentQueriesLength =
+              mockSuggestions.length + exampleRecentQueries.length;
+
+            expect(suggestionOptionLabels).not.toBeNull();
+            expect(suggestionOptionLabels.length).toEqual(
+              suggestionsAndRecentQueriesLength
+            );
+
+            suggestionOptionLabels.forEach((suggestion, index) => {
+              expect(suggestion.title).toEqual(
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:294]: is there a particular reason why you are testing only the last suggestion in the list?

**Review by mmitiche — APPROVED** [review]
👍

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js (hunk 3 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:234]: @SimonMilord this was not fixed?

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js (hunk 4 above)
Comment (erocheleau) [reply packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:294]: It was fixed here @SimonMilord but not above?

**Review by erocheleau — CHANGES_REQUESTED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js (hunk 4 above)
Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticSearchBoxInput/__tests__/quanticSearchBoxInput.test.js:294]: oops forgot I changed it at 2 places, will address that

**Review by olamothe — APPROVED** [review]

**Review by erocheleau — APPROVED** [review]

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**